### PR TITLE
[5.8] Make the redis queue blocking pop atomic (alternative PR)

### PR DESCRIPTION
**Pull request by halaei**
This is an alternative to #27227. The main difference is in the way these PRs handle some edge cases especially when mixing blocking & non-blocking workers/pushers.
In #27227 non-blocking operations do not interact with notify queue and a script is run to resize the notify queue on every blocking pop.
In this PR non-blocking pushers/workers push/pop from notify queue as well.

### CODE REVIEW

**Comment by halaei** [comment]
@yuloh One retry added if blpop was successful https://github.com/laravel/framework/pull/27336/commits/15d19e7dcb1aada466bf4033b86e4bd3a13b0c60

**Comment by halaei** [comment]
@yuloh sorry check https://github.com/laravel/framework/pull/27336/commits/67883ffc4cff7e61c65ed45c54f72e325ebe96d8

**Comment by halaei** [comment]
It is a bit hard to add a test to prevent a bug like the one in https://github.com/laravel/framework/commit/15d19e7dcb1aada466bf4033b86e4bd3a13b0c60
I think I had one test for that in my former PRs...

**Comment by taylorotwell** [comment]
So, which PR do you two think is better 😆

File: tests/Queue/RedisQueueIntegrationTest.php [hunk 1]
```diff
@@ -65,6 +65,31 @@ public function testExpiredJobsArePopped($driver)
...
+     * @dataProvider redisDriverProvider
+     *
+     * @param $driver
+     *
+     * @throws \Exception
+     */
+    public function testBlockingPop($driver)
+    {
+        $this->tearDownRedis();
+        if ($pid = pcntl_fork() > 0) {
```
Comment (yuloh) [inline tests/Queue/RedisQueueIntegrationTest.php]: I don't think the `pcntl` extension is required.  You might need to check if it's installed and skip the test if it's not.

**Comment by yuloh** [comment]
This PR will perform better while solving most of the issues discussed in #27227.  I will close mine.

File: src/Illuminate/Queue/RedisQueue.php [hunk 2]
```diff
@@ -195,60 +198,43 @@ protected function migrate($queue)
     /**
      * Migrate the delayed jobs that are ready to the regular queue.
      *
-     * @param  string  $from
-     * @param  string  $to
+     * @param  string $from
+     * @param  string $to
```
Comment (francislavoie) [inline src/Illuminate/Queue/RedisQueue.php]: Just a PHPDoc nit: see https://laravel.com/docs/5.7/contributions#coding-style, should have 2 spaces in between.

There's a few spots as well.

**Comment by taylorotwell** [comment]
Would need full upgrade guide / breaking change summary like other PR.

**Comment by halaei** [comment]
@taylorotwell I can't think of a change with a high chance of impact. But I sent a PR to docs: https://github.com/laravel/docs/pull/4965

**Comment by taylorotwell** [comment]
Thanks for your help @yuloh and @halaei

**Pull request merged**